Kitodo-SRU-Import/src/main/java/org/kitodo/sruimport/SRUImport.java

```diff
@@ -0,0 +1,193 @@
+        }
+        // TODO: transform hit to Kitodo internal format using SchemaConverter!
+        return null;
+    }
+
+    @Override
+    public SearchResult search(String catalogId, String field, String term, int rows) {
+        loadOPACConfiguration(catalogId);
+        HashMap<String, String> searchFields = new HashMap<>();
+        searchFields.put(field, term);
+        return search(catalogId, searchFields);
+    }
+
+
+    private SearchResult search(String catalogId, Map<String, String> searchParameters) {
+        // TODO: check how the fields of hits from SRU interfaces can be configured via CQL (need only title and id!)
+        loadOPACConfiguration(catalogId);
+        if (searchFieldMapping.keySet().containsAll(searchParameters.keySet())) {
+
+            // Query parameters for HTTP request
+            LinkedHashMap<String, String> queryParameters = new LinkedHashMap<>(parameters);
+
+            // Search fields and terms of query
+            LinkedHashMap<String, String> searchFieldMap = new LinkedHashMap<>();
+            for (Map.Entry<String, String> entry : searchParameters.entrySet()) {
+                searchFieldMap.put(searchFieldMapping.get(entry.getKey()), entry.getValue());
+            }
+
+            try {
+                // FIXME:
+                // adding "query" to the list of parameters to be formatted by 'URLEncodedUtils.format'
+                // results in the "==" between search fields and terms to become URL encoded and a failed query!
+                //queryParameters.put("query", createSearchFieldString(searchFieldMap));
+                URI queryURL = createQueryURI(queryParameters);
+                return performQuery(queryURL.toString() + "&query=" + createSearchFieldString(searchFieldMap));
+            } catch (URISyntaxException e) {
+                logger.error(e.getLocalizedMessage());
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Collection<Document> getMultipleEntriesById(Collection<String> ids, String catalogId) {
+        return null;
+    }
+
+    private SearchResult performQuery(String queryURL) {
+        try {
+            HttpResponse response = sruClient.execute(new HttpGet(queryURL));
+            if (Objects.equals(response.getStatusLine().getStatusCode(), SC_OK)) {
+                return ResponseHandler.getSearchResult(response);
+            }
+        } catch (IOException e) {
+            logger.error(e.getLocalizedMessage());
+        }
+        return new SearchResult();
+    }
+
+    private Document performQueryToDocument(String queryURL) {
+        try {
+            HttpResponse response = sruClient.execute(new HttpGet(queryURL));
+            if (Objects.equals(response.getStatusLine().getStatusCode(), SC_OK)) {
+                return ResponseHandler.transformResponseToDocument(response);
+            }
+        } catch (IOException e) {
+            logger.error(e.getLocalizedMessage());
+        }
+        return null;
```

Comment:
Null returns should be avoided. Is it possible to return here for e.g. empty Document?

Reply:
Done.